OpenClaw/Engine/Actor/Components/ControllableComponent.cpp

```diff
@@ -1026,8 +1058,27 @@ void ClawControllableComponent::VOnHealthBelowZero(DamageType damageType, int so
 
 void ClawControllableComponent::VOnHealthChanged(int32 oldHealth, int32 newHealth, DamageType damageType, Point impactPoint, int sourceActorId)
 {
+    // If claw still has health
+    if (newHealth <= 0) {
+        return;
+    } 
+
+    if (oldHealth == newHealth) {

```

Comment:
I would just make this one "if", like:

if (damageType == DamageType_SirenProjectile)
{
    m_pClawAnimationComponent->SetAnimation("freeze")
    AddFrozenTime(1);
    m_State = ClawState_Frozen;
    // Since this is abit hacky, return as soon as possible into the method
    return;
}